docs/deployment/private/self-hosted-deployment.mdx

```diff
@@ -0,0 +1,272 @@
+:::
+
+## Prerequisites
+
+Before continuing, ensure you go through the following checklist and confirm that you meet all the requirements
+
+- [Docker v2.27.1](/getting-started/build/prerequisites/#step-3-install-docker-compose-v2271-or-greater) (Or greater)
+- [DDN CLI](/getting-started/build/prerequisites#step-1-install-and-authorize-the-hasura-cli) (Latest)
+- [Helm3](https://helm.sh/docs/intro/install/) (Prefer latest)
+- [Hasura VS Extension](/getting-started/build/prerequisites#install-lsp) (Recommended, but not required)
+- Access to a Kubernetes cluster
+- Ability to build and push images that can then be pulled down from the Kubernetes cluster
+- A user account on the Hasura DDN Control Plane
+- A data plane entry created by the Hasura team.  This will be referenced as `<ddn-id>`
+- A data plane id & key created by the Hasura team.  These will be referenced as `<data-plane-id>` and `<data-plane-key>`
+
+## Step 1. Local machine development
+
+```bash title="Login with the DDN CLI."
+ddn auth login
+```
+
+```bash title="Create a local directory where your supergraph will be created.  Afterwards, cd into it."
+mkdir hasura_project && cd hasura_project
+```
+
+```bash title="Initialize supergraph."
+ddn supergraph init .
+```
+
+```bash title="Add a connector.  Here, we are using the -i flag for interactive mode."
+ddn connector init -i
+```
+
+```bash title="Introspect the data source.  Substitute <connector-name> with name you chose for your connector above."
+ddn connector introspect <connector-name>
+```
+
+```bash title="Add models, commands and relationships based on the output from previous command.  Substitute <connector-name> with name you chose for your connector above."
+ddn model add <connector-name> "*"
+ddn command add <connector-name> "*"
+ddn relationship add <connector-name> "*"
+```
+
+```bash title="Build the supergraph locally."
+ddn supergraph build local
+```
+
+```bash title="Run docker.  Ensure all containers start up successfully."
+ddn run docker-start
+```
+
+```bash title="Verify via local console."
+ddn console --local
+```
+
+At this point, you have connected and introspected a data source and built your supergraph locally.  Verify that everything is working as expected before moving on to the next section.
+
+## Step 2. Create a cloud project and create a build
+
+Ensure that you are running the commands here from the root of your supergraph.
+
+```bash title="Create cloud project."
+ddn project init --data-plane-id <ddn-id>
+```
+
+This command will create a cloud project and will report back as to what the name of your cloud project is.  We will reference this name going forward as `<ddn-project-name>`.
+
+Next, you will need to access the `.env.cloud` file which gets generated at the root of your supergraph.
```

Comment:
For this step, we need to have connectors deployed, right? I think we should move this to a later step.

Reply:
Yup, will make the change!